crates/phactory/src/contracts/support.rs

```diff
@@ -567,5 +569,130 @@ fn local_cache_ops() -> sidevm::DynCacheOps {
     &CacheOps
 }
 
+#[instrument(skip_all, fields(id=%ShortId(id)), name = "run")]
+pub fn block_on_run_module(
+    id: VmId,
+    module: &WasmModule,
+    args: Vec<String>,
+    timeout: Duration,
+    sidevm_event_tx: OutgoingRequestChannel,
+    log_handler: impl Fn(VmId, u8, String),
+) -> Result<JsValue> {
+    info!("Run wasm module timeout={}ms", timeout.as_millis());
+    enum OutMsg {
+        Log(u8, String),
+        Output(Vec<u8>),
+        Terminated,
+        Timeout,
+        Error(anyhow::Error),
+    }
+    // Channel to bridge between the sync and async world
+    let (output_tx, output_rx) = mpsc::channel();
+    let (event_tx, mut event_rx) = tokio::sync::mpsc::channel(1);
+    let tx_for_logging = output_tx.clone();
+    let config = WasmInstanceConfig {
+        max_memory_pages: 256, // 16MB
+        gas_per_breath: SidevmConfig::default().vital_capacity,
+        cache_ops: local_cache_ops(),
+        scheduler: None,
+        weight: 0,
+        id,
+        event_tx,
+        log_handler: Some(Box::new(move |_vmid, level, msg| {
+            if let Err(err) = tx_for_logging.send(OutMsg::Log(level, msg.into())) {
+                error!("Failed to send log message to response channel: {}", err);
+            }
+        })),
+    };
+    let (mut wasm_run, _env) = module
+        .run(args, config)
+        .context("Failed to start sidevm instance")?;
+    tokio::spawn(
+        async move {
+            /// Returns true if the sidevm should be terminated
+            async fn forward_event(
+                vmid: VmId,
+                event: sidevm::OutgoingRequest,
+                response_tx: &mpsc::Sender<OutMsg>,
+                event_tx: &OutgoingRequestChannel,
+            ) -> bool {
+                match event {
+                    event @ sidevm::OutgoingRequest::Query { .. } => {
+                        match event_tx.send((vmid, event)).await {
+                            Ok(_) => false,
+                            Err(err) => {
+                                error!("Failed to send query request: {err}");
+                                true
+                            }
+                        }
+                    }
+                    sidevm::OutgoingRequest::Output(output) => {
```

Comment:
Since the `Output` is only handled here, do you think `handle_sidevm_ocall()` in `lib.rs` can be renamed to `handle_sidevm_query_ocall()`?